aptos-move/framework/aptos-framework/sources/permissioned_signer.move

```diff
@@ -0,0 +1,629 @@
+        );
+    }
+
+    /// =====================================================================================================
+    /// StoredPermission operations
+    ///
+    /// check if StoredPermission has at least `threshold` capacity.
+    fun is_above(perm: &StoredPermission, threshold: u256): bool {
+        match (perm) {
+            StoredPermission::Capacity(capacity) => *capacity > threshold,
+            StoredPermission::Unlimited => true,
+        }
+    }
+
+    /// consume `threshold` capacity from StoredPermission
+    fun consume_capacity(perm: &mut StoredPermission, threshold: u256): bool {
+        match (perm) {
+            StoredPermission::Capacity(current_capacity) => {
+                if (*current_capacity >= threshold) {
+                    *current_capacity = *current_capacity - threshold;
+                    true
+                } else { false }
+            }
+            StoredPermission::Unlimited => true
+        }
+    }
+
+    /// increase `threshold` capacity from StoredPermission
+    fun increase_capacity(perm: &mut StoredPermission, threshold: u256) {
+        match (perm) {
+            StoredPermission::Capacity(current_capacity) => {
+                *current_capacity = *current_capacity + threshold;
+            }
+            StoredPermission::Unlimited => (),
+        }
+    }
+
+    /// merge the two stored permission
+    fun merge(lhs: &mut StoredPermission, rhs: StoredPermission) {
+        match (rhs) {
+            StoredPermission::Capacity(new_capacity) => {
+                match (lhs) {
+                    StoredPermission::Capacity(current_capacity) => {
+                        *current_capacity = *current_capacity + new_capacity;
+                    }
+                    StoredPermission::Unlimited => (),
+                }
+            }
+            StoredPermission::Unlimited => *lhs = StoredPermission::Unlimited,
+        }
+    }
+
+    /// =====================================================================================================
+    /// Permission Management
+    ///
+    /// Authorizes `permissioned` with the given permission. This requires to have access to the `master`
+    /// signer.
+
+    inline fun map_or<PermKey: copy + drop + store, T>(
+        permissioned: &signer,
+        perm: PermKey,
+        mutate: |&mut StoredPermission| T,
+        default: T,
+    ): T {
+        let permission_signer_addr = permission_address(permissioned);
+        assert!(
+            exists<PermissionStorage>(permission_signer_addr),
+            error::permission_denied(E_NOT_ACTIVE)
+        );
+        let perms =
+            &mut borrow_global_mut<PermissionStorage>(permission_signer_addr).perms;
+        let key = copyable_any::pack(perm);
+        if (big_ordered_map::contains(perms, &key)) {
+            let value = perms.remove(&key);
+            let return_ = mutate(&mut value);
+            perms.upsert(key, value);
```

Comment:
since you are removing, you can just call `add`

Reply:
Done